extensions/ql-vscode/src/databases/local-databases/database-item-impl.ts

```diff
@@ -121,7 +121,9 @@ export class DatabaseItemImpl implements DatabaseItem {
   /**
    * Returns information about a database.
    */
-  private async getDbInfo(server: cli.CodeQLCliServer): Promise<cli.DbInfo> {
+  private async getDbInfo(
+    server: Pick<cli.CodeQLCliServer, "resolveDatabase">,
```

Comment:
Why do we have to do this? Can we avoid it? I'm worried it'll hurt readability/maintainability gradually.